### PR TITLE
Upgrade jsonschema library to 5.2.0

### PR DESCRIPTION
**Pull request by s-l-teichmann**
The jsonschema library introduced a subtle change when compiling schemas with regard of handling external sources. It may be a bug.

The resulting problem can be circumvented by simplifying the usage of
the schema compiler. 

Partly solves #345

### CODE REVIEW

**Review by JanHoefelmeyer — APPROVED** [review]
Works for me

File: .github/workflows/itest.yml [hunk 1]
```diff
@@ -11,11 +11,17 @@ jobs:
         with:
           go-version: 1.19.1
 
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
```
Comment (tschmidtb51) [inline .github/workflows/itest.yml:18]: Do we need the setup node action here as we install Node.js manually in https://github.com/csaf-poc/csaf_distribution/pull/349/files#diff-9e14b99a1dd4003f9986a20557f9e3ddbd8d071fc0d62d3e3eec7bf5bf2c0244

**Review by tschmidtb51 — COMMENTED** [review]
A small comment. I think having the part in the script helps for easy local tests...

File: .github/workflows/itest.yml (hunk 1 above)
Comment (tschmidtb51) [reply .github/workflows/itest.yml:18]: Flagging @s-l-teichmann to answer the question...

Comment (s-l-teichmann) [reply .github/workflows/itest.yml:18]: This was contributed by @bernhardreiter to make the integration test work again. Bernhard, what's the deal?

Comment (bernhardreiter) [reply .github/workflows/itest.yml:18]: The setup is improved to follow the original pattern:
 * .github/workflows/itest.yml  uses the github actions, which allow caching on the github runner's side.
 * docs/scripts/prepareUbuntuInstanceForITests.sh is used to prepare a naked Ubuntu system once. And is not part of the testing script itself. 
 
This allows the setup steps to be done only once and not per test run on a VM. It makes both the automatic and the manual tests faster (and saves a tiny bit of genery. ;) )

**Review by bernhardreiter — APPROVED** [review]
The integration test ran fine. 
Briefly reading over the changes, there is no obvious problem.
LGTM

**Pull request merged**